frontend/src/metabase/query_builder/utils.ts

```diff
@@ -39,3 +58,15 @@ export function getURLForCardState({ card }, dirty, query = {}, objectId) {
   }
   return Urls.question(card, options);
 }
+
+export function isQuestionEdited(
+  question: Question | null,
+  originalQuestion: Question | null,
+): originalQuestion is Question {
+  return (
+    question != null &&
+    !question.isSaved() &&
+    originalQuestion != null &&
```

Comment:
This is the logic that's extracted from `frontend/src/metabase/selectors/app.ts` and `frontend/src/metabase/containers/SaveQuestionModal.tsx`. They're technically the same.

I think we need to rely on `originalQuestion.isDataset()` because when you edit a model it won't prompt you to replace the model, unlike editing a question which will ask you to either modify the saved question or save it as a new question instead. i.e. if we only check `question.isSave()` we will ask users whether they want to save a modified model but we don't want that this is what we should show
![image](https://github.com/metabase/metabase/assets/1937582/c24d7463-ac8b-4b85-a377-b4b9c037f0d2)

The same thing applies to the question lineage we won't show question lineage for models
![image](https://github.com/metabase/metabase/assets/1937582/880b0f3a-08fb-4cb5-96c7-dc493bff6a93)

This logic is extracted here for the purpose of those 2 use cases, but it is used in this fix as well.

Now I've explained the reason that this logic exists. And the reason it's used for this fix too is that we need only want to show the browser prompt when users are [editing existing native questions](https://github.com/metabase/metabase/pull/30477). If we only rely on `question.isSave()` solely we will show the browser warning for new native questions too (`+New` > `SQL Query`) which we don't want.

I hope this is clear